index.js

```diff
@@ -19,39 +19,45 @@ function parse(str) {
 function getPath(opts) {
 	opts = opts || {};
 
-	if (process.platform === 'win32') {
-		return '';
-	}
-
+	// HISTFILE might be cross-platform, so win32 check is after!
 	if (process.env.HISTFILE) {
 		return process.env.HISTFILE;
 	}
 
-	const homeDir = os.homedir();
-
-	const paths = new Set([
-		path.join(homeDir, '.bash_history'),
-		path.join(homeDir, '.zsh_history'),
-		path.join(homeDir, '.history')
-	]);
-
+	// Add extra paths from opts
+	const paths = new Set([]);
 	if (opts.extraPaths) {
 		for (const path of opts.extraPaths) {
 			paths.add(path);
 		}
 	}
 
+	// handle *nix history files
+	if (process.platform !== 'win32') {
+		const homeDir = os.homedir();
+		paths.add(path.join(homeDir, '.bash_history'));
+		paths.add(path.join(homeDir, '.zsh_history'));
+		paths.add(path.join(homeDir, '.history'));
+	}
+
 	return Array.from(paths)
 		.filter(fs.existsSync)
 		.map(fp => ({fp, size: fs.statSync(fp).size}))
-		.reduce((a, b) => a.size > b.size ? a : b).fp;
+		.reduce((a, b) => a.size > b.size ? a : b, {
+			fp: null
```

Comment:
Not clear why this change is required.

Reply:
`reduce` was throwing an error because of how Windows is being handled - Win platforms don't resolve early, instead, there are no paths to `reduce`, thus causing an error, or requiring a default value.

IMO, after filtering and mapping, the first path should be used, which can allow the use of:
```js
Array.from(paths).filter(fs.existsSync)[0]
```
instead of mapping and reducing as well. If index 0 doesn't exist, it returns undefined which is handled [here](https://github.com/sindresorhus/shell-history/pull/5/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R56).